### PR TITLE
Docker upgrade

### LINKED ISSUE #739: Docker 1.12 is getting EOL'd in November
is it possible to use docker-ce instead of os distribution docker on ubuntu 16.04?

### PR DESCRIPTION
**Pull request by emmetthitz**
Exactly what it looks like

Fixes #739

### CODE REVIEW

File: ansible/group_vars/all.yaml [hunk 1]
```diff
@@ -2,8 +2,8 @@
 # VERSIONS
 kubernetes_yum_version: 1.8.0-0
 kubernetes_deb_version: 1.8.0-00
-docker_engine_yum_version: 1.12.6-1.el7.centos
-docker_engine_apt_version: 1.12.6-0~ubuntu-xenial
+docker_engine_yum_version: 17.03.2.ce-1.el7.centos
+docker_engine_apt_version: 17.03.1~ce-0~ubuntu-xenial
```
Comment (dkoshkin) [inline ansible/group_vars/all.yaml]: We cant use 2 different versions, especially since `17.03.1` is not the one thats tested by kubernetes.

Comment (emmetthitz) [reply ansible/group_vars/all.yaml]: There is no 17.03.2 up on the ubuntu repositories though.

Comment (dkoshkin) [reply ansible/group_vars/all.yaml]: Take a look at https://download.docker.com/linux/

Comment (emmetthitz) [reply ansible/group_vars/all.yaml]: I lied.

**Comment by swade1987** [comment]
@emmetthitz what is the latest status on this?

**Comment by swade1987** [comment]
when is this likely to be merged?

**Comment by emmetthitz** [comment]
@dkoshkin LGTM

File: ansible/roles/packages-docker/tasks/main.yaml [hunk 2]
```diff
@@ -1,9 +1,8 @@
 ---
   # YUM
-  - name: install docker-engine yum package
-    yum:
-      name: docker-engine-{{ docker_engine_yum_version }}
-      state: present
+  # Can't use yum module as --setopt=obsoletes=0 is not supported
```
Comment (alexbrand) [inline ansible/roles/packages-docker/tasks/main.yaml]: Might be useful to document why we need this flag.

**Comment by emmetthitz** [comment]
Is there a reason we don't want to ship this with 1.7?

**Comment by emmetthitz** [comment]
Or is it just the validation work that needs to be done?

**Comment by swade1987** [comment]
We can't really be releasing new versions of KET with versions of Docker which are EOL

**Comment by dkoshkin** [comment]
Ran a long test https://circleci.com/gh/apprenda/kismatic/1368 for this PR, this is ready to go.

**Pull request merged**